### PR TITLE
fix(golangci-lint): fix errors as raised by linter

### PR DESCRIPTION
**Pull request by boreyuk**
Linked to #320

### CODE REVIEW

**Comment by YOU54F** [comment]
thanks @boreyuk appreciate the contribution and response on the issue. I've tagged Matt and let the PR run.

I've also updated the release pipeline to 1.20 so this linter probably wants to run against that as well. (I think we can probably drop 1.17/1.18 as they are unsupported for go, but that probably warrants another PR)

**Comment by mefellows** [comment]
> 2023-07-19T14:53:27.3368767Z fatal error: non-Go code set up signal handler without SA_ONSTACK flag

This relates to #269, I think. We have yet to chase down that nasty bug.

**Comment by mefellows** [comment]
This is great, thank you. I think for the test files and the unused functions, I'll take those as is. For the real ones where we swallow the errors, perhaps we actually need to `panic` (dramatic, but there is no other pathway in many of them) or at the very least, log an `ERROR` level statement. Thoughts based on the contexts?

**Comment by boreyuk** [comment]
I agree - i think that it's better to explicitely panic than silently fail 😛 
I will add another commit with the `ERROR` logs and the `panic`s for unrecoverable errors.

**Comment by YOU54F** [comment]
thanks for updated commit @boreyuk have kicked off the workflow now

**Comment by YOU54F** [comment]
So that is a plugin specific error (the plugins themselves define the content make up of their messages, and can choose to error)

I think that error will be coming from the underlying plugin we are calling. 

As we can't proceed in that case, and we need to perform the action in the message provided, it would make sense for this to be an error for the user.

the plugins communicate with the central pact framework via grpc so there is room for error here for a number of variables.

Having this swallowed here I think would be disadvantageous all round, so given that I would suggest a panic (provided it that also give us that detail) 

Thanks for having a look again my friend, really do appreciate it

**Comment by boreyuk** [comment]
Thanks for your feedback ! I just modified the test in order to make it pass, and I commited before seeing your comment  :sweat_smile:
I can add another test to check if we "correctly panic" if we have an error on the plugin - anyway, the workflows should pass now :smile:

**Comment by YOU54F** [comment]
No worries @boreyuk I'll let @mefellows make the final call, as he is the core maintainer of the repo.

**Comment by YOU54F** [comment]
I think we should drop 1.17 due it being out of support

https://endoflife.date/go

is there value is testing across the different supported versions of go with golanglint-ci or just the latest version we use to build/release

File: utils/deepcopy.go
Comment (YOU54F) [inline utils/deepcopy.go:11]: was this util func just unused?

**Comment by YOU54F** [comment]
I think we need to skip linting of the C directive in `internal/native/lib.go` or maybe there is an option to allow it.

either way it works, once built so I assume this is a bit of a red herring?

File: utils/deepcopy.go
Comment (boreyuk) [reply utils/deepcopy.go:11]: Yep, at least golangci-lint flagged it as unused and I couldn't find any reference/usage to this function

**Comment by boreyuk** [comment]
> I think we need to skip linting of the C directive in `internal/native/lib.go` or maybe there is an option to allow it.
> 
> either way it works, once built so I assume this is a bit of a red herring?

I added a commit which modifies the golangci-lint in order to ignore the `internal/native/lib.go`. I'm not familiar with configuring golangci-lint through Github Actions, so I hope this will work!

**Comment by YOU54F** [comment]
> I'm not familiar with configuring golangci-lint through Github Actions, so I hope this will work!

😄 s'all good, nothing like a bit of experimentation 👍🏾

**Review by YOU54F — APPROVED** [review]
Awesome! 

Man it will be nice to be able to see green in the pipeline again. Thanks dude

**Review by mefellows — APPROVED** [review]

**Comment by mefellows** [comment]
This is great, thanks. 

Those additional TODOs are also good food for thought, but we can pick them up in the future.

**Comment by boreyuk** [comment]
Nice ! Thanks to you two 🎉

**Comment by mefellows** [comment]
You're so welcome, we really appreciate it. If you're up for me, I'd love to chat with you (just hit me up in the #pact-go channel in slack.pact.io).

**Pull request merged**
